pkg/activator/net/throttler_test.go

```diff
@@ -251,8 +257,8 @@ func TestThrottlerWithError(t *testing.T) {
 				time.Sleep(200 * time.Millisecond)
 			}
 
-			tryContext, cancel := context.WithTimeout(context.TODO(), 100*time.Millisecond)
-			defer cancel()
+			tryContext, cancel2 := context.WithTimeout(context.TODO(), 100*time.Millisecond)
```

Comment:
I don't think you need a new variable here.

Reply:
I tried that but then the test hangs in `waitInformers()` since `cancel` got overwritten, `tryContext` gets canceled twice and `ctx.Done()` never gets closed.